text/0249-v2-experiments.md

```diff
@@ -0,0 +1,127 @@
+constructs, which do not provide them with the benefits of higher level abstractions.
+
+This RFC proposes that we stop experimenting in the main repository. To keep the implementation simple and to avoid
+possible inadvertent breakages, especially considering the need to convert from Typescript into a variety of languages,
+we should not use any clever techniques to simply hide the experiments. The code should be all stable. A user that
+installs aws-cdk-lib using npm or pip or any other package manager should be confident that there will be no intentional
+breaking changes in the 2.x line of releases for its lifetime.
+
+# README
+
+The following section will be removed from the CDK Readme:
+
+> Modules in the AWS Construct Library are designated Experimental while we build them; experimental modules may have
+> breaking API changes in any release. After a module is designated Stable, it adheres to semantic versioning, and only
+> major releases can have breaking changes. Each module's stability designation is available on its Overview page in the
+> AWS CDK API Reference. For more information, see Versioning in the CDK Developer Guide.
+
+The existing contributor guide does not mention experimental code, so the following will be added after the intro
+paragraph:
+
+> If you are contributing to a 2.x release, keep in mind that you may not introduce breaking changes to any public
+> facing API. Experimental modules are developed in a separate repository. [link]
+
+The new aws-cdk-previews repository will have the following Readme:
+
+> This repository contains experimental code related to the AWS Cloud Development Kit (AWS CDK). CDK is an open-source
+> software development framework to define cloud infrastructure in code and provision it through AWS CloudFormation.
+> Please refer to the main repository here [link]. [TODO...]
+
+The contributor guide can be copied from the main repository, since the toolchain for building the preview repository
+will be basically the same.
+
+# Design Summary
+
+- Upon release of v2.0 to GA, all experimental modules will be automatically marked as stable (in fact, the metadata in
+  the code regarding stability can be removed, since everything will be stable). This means that the opportunity for the
+  team and for the community to influence the shape of existing experimental APIs is between now and then. Once we go
+  GA, we have to live with our design decisions.
+- Any APIs that cannot be graduated with confidence in time for the v2 release will be deprecated in 1.x, removed from
+  the v2 branch, and moved to the preview repository. This option should be minimized, since some customers may have
+  deployed production stacks that depend on these APIs.
+- Create a new repository called aws-cdk-previews that will be the new home for experimental development and developer
+  preview releases. The packages in this repository will be vended in the same way that 1.x packages are vended today,
```

Comment:
> The packages in this repository will be vended in the same way that 1.x packages are vended today,

I don't understand why would we vend these as separate packages? This brings back the same dependency problem that we started v2.0 to solve...

Why not one `aws-cdk-lib-experimental` library?

Reply:
This was more about reducing effort with the transition. I'm open to a single package, but I don't know how much extra work that would entail.